src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -1238,7 +1233,7 @@ private Action<Diagnostic> GetAddDiagnostic(ISymbol contextSymbol, ImmutableArra
         {
             return diagnostic =>
             {
-                if (isGeneratedCodeSymbol && shouldSuppressGeneratedCodeDiagnostic(diagnostic, analyzer, compilation, cancellationToken))
+                if (shouldSuppressGeneratedCodeDiagnostic(diagnostic, analyzer, compilation, cancellationToken))
```

Comment:
I think right now, common case (diagnostic not in generated code) is expensive to calculate and rare case (diagnostics in generated code) is expensive.